recursive_art.py

```diff
@@ -42,9 +63,48 @@ def evaluate_random_function(f, x, y):
         -0.5
         >>> evaluate_random_function(["y"],0.1,0.02)
         0.02
+        >>> evaluate_random_function(["prod"],4,2)
+        8
+        >>> evaluate_random_function(["prod", ["x"], ["y"]], 0, 0)
+        0
+        >>> evaluate_random_function(["prod", ["avg", ["x"],["y"]], ["x"]], 2, 2)
+        4
+        >>> evaluate_random_function(["avg", ["prod", ["avg", ["x"],["y"]], ["x"]], ["prod", ["x"], ["y"]]], 0, 0)
+        0.0
+        >>> evaluate_random_function(["sin_pi", ["avg", ["prod", ["avg", ["x"],["y"]], ["x"]], ["prod", ["x"], ["y"]]]], 0, 0)
+        0.0
+        >>> evaluate_random_function(['sin_pi', ['cos_pi', ['cos_pi', ['x']]]], -1, -1)
+        0.0
     """
-    # TODO: implement this
-    pass
+    if len(f) == 2:
+        evaluate_random_function(f[0],evaluate_random_function(f[1],x,y),y)
+    if len(f) == 3:
+        evaluate_random_function(f[0],
+                                 evaluate_random_function(f[1],x,y),
+                                 evaluate_random_function(f[2],x,y))
+    if f == ["x"]:
+        # print("return : ", x)
+        return x
+    elif f == ["y"]:
+        # print("return : ", y)
+        return y
+    elif f[0]  == "prod":
+        return x * y
+    elif f[0] == "cos_pi":
+        # print("return : ", math.cos(math.pi * f[1]))
+        return math.cos(math.pi * x)
+    elif f[0] == "sin_pi":
+        # print("return : ", math.sin(math.pi * f[1]))
+        return math.sin(math.pi * x)
+    elif f[0] == "avg":
+        # print("return : ", (f[1] + f[2] / 2))
+        return (x + y) / 2
+    elif f[0] == "square":
+        # print("return : ", f[1]**2 )
+        return x ** 2
+    elif f[0] == "cube":
+        # print("return : ", f[1]**3 )
+        return x ** 3
```

Comment:
Great job overall, but I think you can remove the commented print code when submitting